JavaScript/JavaScriptSDK/Util.ts

```diff
@@ -580,19 +580,28 @@ module Microsoft.ApplicationInsights {
         /**
         * Checks if a request url is not on a excluded domain list and if it is safe to add correlation headers
         */
-        public static canIncludeCorrelationHeader(config: IConfig, requestUrl: string) {
+        public static canIncludeCorrelationHeader(config: IConfig, requestUrl: string, currentHost: string) {
             if (config && config.disableCorrelationHeaders) {
                 return false;
             }
 
+            if (!requestUrl) {
+                return false;
+            }
+
+            let requestHost = UrlHelper.parseUrl(requestUrl).host.toLowerCase();
+            if (config && !config.enableCorsCorrelation && requestHost !== currentHost) {
```

Comment:
This should be if((config && !config.enableCorsCorrelation) || requestHost !== currentHost) // either should return false

Reply:
no, it should be && => we should proceed (not return false) if hosts don't match but enableCorsCorrelation is true.